module/bdev/aio/bdev_aio.c

```diff
@@ -232,12 +241,32 @@ bdev_aio_writev(struct file_disk *fdisk, struct spdk_io_channel *ch,
 }
 
 static void
-bdev_aio_flush(struct file_disk *fdisk, struct bdev_aio_task *aio_task)
+bdev_aio_flush(void *arg)
 {
-	int rc = fsync(fdisk->fd);
+	struct aio_request_ctx *ctx = arg;
 
-	spdk_bdev_io_complete(spdk_bdev_io_from_ctx(aio_task),
-			      rc == 0 ? SPDK_BDEV_IO_STATUS_SUCCESS : SPDK_BDEV_IO_STATUS_FAILED);
+	SPDK_NOTICELOG("[flush]\n");
+
+	ctx->status = fsync(ctx->fdisk->fd);
+	ctx->errnum = (ctx->status < 0) ? errno : 0;
+}
+
+static void
+bdev_aio_unmap(void *arg)
+{
+	struct aio_request_ctx *ctx = arg;
+	uint32_t block_size;
+	uint64_t range[2];
+
+	block_size = ctx->fdisk->disk.blocklen;
+
+	range[0] = ctx->range[0] * block_size;
+	range[1] = ctx->range[1] * block_size;
+
+	SPDK_NOTICELOG("[unmap] range=%lu,%lu\n", range[0], range[1]);
```

Comment:
Should use PRIu64 instead of lu for portability.

Reply:
Quite probably, but %lu is already used in lots of other places as well.